src/main/java/org/broadinstitute/hellbender/engine/spark/SAMRecordToGATKReadAdapterSerializer.java

```diff
@@ -0,0 +1,97 @@
+package org.broadinstitute.hellbender.engine.spark;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.Serializer;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import htsjdk.samtools.*;
+import org.broadinstitute.hellbender.utils.read.SAMRecordToGATKReadAdapter;
+
+public class SAMRecordToGATKReadAdapterSerializer extends Serializer<SAMRecordToGATKReadAdapter> {
+
+    private BAMRecordCodec lazyCodec = new BAMRecordCodec(null, new LazyBAMRecordFactory());
+
+    @Override
+    public void write(Kryo kryo, Output output, SAMRecordToGATKReadAdapter adapter) {
+        SAMRecord record = adapter.getSamRecord();
+        // serialize reference names to avoid having to have a header at read time
+        output.writeString(record.getReferenceName());
+        output.writeString(record.getMateReferenceName());
+        lazyCodec.setOutputStream(output);
+        lazyCodec.encode(record);
+
+        // clear indexing bin after encoding to ensure all SAMRecords compare properly
+        record.setFlags(record.getFlags());
```

Comment:
Because calling encode will mutate the record (set the indexing bin field) which has implications for equality.